src/Themes/ui/ColorPicker/index.tsx

```diff
@@ -0,0 +1,239 @@
+import { ColorPreview } from "./ColorPreview";
+import { HexInput, formatRGBtoHEX } from "./HexInput";
+import { HueSlider } from "./HueSlider";
+import { RGBDigit } from "./RGBDigit";
+import { SVCanvas } from "./SVCanvas";
+import React, { Dispatch, useMemo, useReducer, useState } from "react";
+import { useTheme } from "@mui/material/styles";
+import { Modal } from "../../../ui/React/Modal";
+import Button from "@mui/material/Button";
+
+export type HSV = {
+  hue: number;
+  sat: number;
+  val: number;
+};
+
+export type RGB = {
+  r: number;
+  g: number;
+  b: number;
+  a?: number;
+};
+
+function normalize(color: HSV | RGB) {
+  const isRGB = (c: RGB | HSV): c is RGB => Object.hasOwn(color, "r");
+  const rgb: RGB = isRGB(color) ? color : HSVtoRGB(color);
+  const hsv: HSV = !isRGB(color) ? color : RGBtoHSV(color);
+  return { rgb, hsv };
+}
+
+type OpenColorPickerButtonProps = {
+  color: string;
+  title: string;
+  onColorChange: (color: string) => void;
+};
+
+function parseHEXtoRGB(hex: string): RGB {
+
+  if (hex.length == 4) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[1]}`, 16),
+      g: Number.parseInt(`${hex[2]}${hex[2]}`, 16),
+      b: Number.parseInt(`${hex[3]}${hex[3]}`, 16),
+    };
+  }
+
+  if (hex.length == 7 || hex.length == 9) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[2]}`, 16),
+      g: Number.parseInt(`${hex[3]}${hex[4]}`, 16),
+      b: Number.parseInt(`${hex[5]}${hex[6]}`, 16),
+      a: Number.parseInt(`${hex[7]}${hex[8]}`, 16),
+    };
+  }
+
+  throw new Error("Invalid hex string");
```

Comment:
If it shouldn't be reached then make sure it _cannot_ be reached. Your last branch is checking length 7 and 9. What happens on the alpha check for hex[8] if the length is 7?

You can make the check exhaustive by padding the string with zeros then clamping it to length 9. That removes the need for the 2nd if (and error throw).

```suggestion
function parseHEXtoRGB(hex: string): RGB {
  if (hex.length === 4) {
    return {
      r: Number.parseInt(`${hex[1]}${hex[1]}`, 16),
      g: Number.parseInt(`${hex[2]}${hex[2]}`, 16),
      b: Number.parseInt(`${hex[3]}${hex[3]}`, 16),
    };
  }
  
  hex = hex.padEnd(9, 'FF');
  return {
    r: Number.parseInt(`${hex[1]}${hex[2]}`, 16),
    g: Number.parseInt(`${hex[3]}${hex[4]}`, 16),
    b: Number.parseInt(`${hex[5]}${hex[6]}`, 16),
    a: Number.parseInt(`${hex[7]}${hex[8]}`, 16),
  };
```

Reply:
that could potentially lead to garbage if the hex string is actually wrongly formatted.
This function just parses a string to an RGB color, ensuring a correct input should be done wherever it is used. 
This function _must_ throw when the input is wrongly formatted and the consumer of this function _must_ handle that case.
In our case this error is implicitly handled by the input validation for both the HexInput and onColorChange function


I agree on the first part tho, handling length 7 vs 9 should be done separately. That was an oversight on my part

Reply:
Discussing offline in Discord. Expect some small nudges here.